### PR TITLE
#907 Adds oauth token as param to codebuild.GitHubEnterpriseSource

### PR DESCRIPTION
**Pull request by eaddingtonwhite**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

Addresses #907 

cc/ @moofish32

### CODE REVIEW

File: packages/@aws-cdk/aws-codebuild/lib/source.ts [hunk 1]
```diff
@@ -92,18 +92,20 @@ export class GitHubSource extends BuildSource {
 }
 
 /**
- * GitHub Enterprice Source definition for a CodeBuild project
+ * GitHub Enterprise Source definition for a CodeBuild project
  */
 export class GitHubEnterpriseSource extends BuildSource {
-  constructor(private readonly cloneUrl: string) {
+  constructor(private readonly cloneUrl: string, private readonly oauthToken: any) {
```
Comment (ellery44) [inline packages/@aws-cdk/aws-codebuild/lib/source.ts]: Was not sure if I should leave make oauthToken optional?
```typescript
constructor(private readonly cloneUrl: string, private readonly oauthToken?: any) {
```

I was mostly copying how it was done w/ `GitHubSource` but now this is breaking change I believe.

Comment (moofish32) [reply packages/@aws-cdk/aws-codebuild/lib/source.ts]: @skinny85 - should the token be `any`?

Comment (skinny85) [reply packages/@aws-cdk/aws-codebuild/lib/source.ts]: Going by the [GitHub CodePipeline Action example](https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-codepipeline/lib/github-source-action.ts#L44), it seems like `oauthToken` should be a `cdk.Secret`.

Comment (ellery44) [reply packages/@aws-cdk/aws-codebuild/lib/source.ts]: @skinny85 @moofish32 I was going off. https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-codebuild/lib/source.ts#L79

Do you want me to change both to `cdk.Secret`?

File: packages/@aws-cdk/aws-codebuild/lib/source.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ export class CodePipelineSource extends BuildSource {
  * GitHub Source definition for a CodeBuild project
  */
 export class GitHubSource extends BuildSource {
-  constructor(private readonly httpscloneUrl: string, private readonly oauthToken: any) {
+  constructor(private readonly httpscloneUrl: string, private readonly oauthToken: cdk.Secret) {
```
Comment (eladb) [inline packages/@aws-cdk/aws-codebuild/lib/source.ts]: Can we convert this to a props interface?

**Comment by eaddingtonwhite** [comment]
@eladb let me know if there is anything else you want changed. 👍

**Pull request merged**